src/esm_tools/cli.py

```diff
@@ -1,24 +1,60 @@
 """
 Functionality for displaying the version number
 """
+import shutil
 import sys
+
 import click
-import esm_tools
+
 import esm_tests
+import esm_tools
 
 # click.version_option read the PKG_INFO which contains the wrong version
 # number. Get it directly from __init__.py
 version = esm_tools.__version__
 
+
+@click.group()
 @click.version_option(version=version)
-@click.option("--test-state", is_flag=True, help="Prints the state of the last tested experiments.")
-@click.command()
-def main(test_state):
-    """Console script for esm_tools"""
+def main():
+    pass
+
+
+@main.command()
+def test_state():
+    """Prints the state of the last tested experiments."""
+
+    esm_tests.test_utilities.print_state_online()
+
+    return 0
 
-    if test_state:
-        esm_tests.test_utilities.print_state_online()
 
+@main.command()
+@click.option(
+    "-t",
+    "--type",
+    type=click.Choice(["component", "setup"], case_sensitive=False),
+    help="Creates either a new component (default) or a new setup",
+    default="component",
+    show_default=True,
+)
+@click.argument("name", nargs=1)
+def create_new_config(name, type):
```

Comment:
What about specifying a particular editor?

Reply:
Click is smart enough that it _should_ try to use your $EDITOR variable if you have it set or ask you otherwise. I don't want to dump people into `vim` that might make some users unhappy.